.github/workflows/cluster_cleanup.yml

```diff
@@ -99,21 +89,75 @@ jobs:
-            az resource delete -v --id $mq_resource --verbose
-          done
+            for mq_resource in $(az resource list -g ${{ env.RESOURCE_GROUP }} --query "[?$in_cluster_ext_loc && $is_mq_child_resource].id" -o tsv); do
+              az resource delete -v --id $mq_resource --verbose
+            done
       - name: Delete MQ instances
-        run: |
-          mq_type="Microsoft.IoTOperationsMQ/mq"
-          in_cluster_ext_loc="contains(to_string(extendedLocation.name), '${{ env.CLUSTER_PREFIX }}')"
-          
-          # MQ instance cannot be deleted until all child resources have successfully deleted
-          sleep 15s
+        uses: azure/cli@v2
+        with:
+          azcliversion: 2.58.0
+          inlineScript: |
+            mq_type="Microsoft.IoTOperationsMQ/mq"
+            in_cluster_ext_loc="contains(to_string(extendedLocation.name), '${{ env.CLUSTER_PREFIX }}')"
+
+            # MQ instance cannot be deleted until all child resources have successfully deleted
+            sleep 15s
 
-          for mq in $(az resource list -g ${{ env.RESOURCE_GROUP }} --resource-type $mq_type  --query "[?$in_cluster_ext_loc].id" -o tsv); do
-            az resource delete -v --id $mq --verbose
-          done
\ No newline at end of file
+            for mq in $(az resource list -g ${{ env.RESOURCE_GROUP }} --resource-type $mq_type  --query "[?$in_cluster_ext_loc].id" -o tsv); do
+              az resource delete -v --id $mq --verbose
+            done
+  asset-cleanup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Az CLI login
+        uses: azure/login@v2
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+      - name: Delete Assets
+        uses: azure/cli@v2
+        with:
+          azcliversion: 2.58.0
+          inlineScript: |
+            asset_type="Microsoft.DeviceRegistry/assets"
+            in_cluster_ext_loc="contains(to_string(extendedLocation.name), '${{ env.CLUSTER_PREFIX }}')"
+            for asset in $(az resource list -g ${{ env.RESOURCE_GROUP }} --resource-type $asset_type --query "[?$in_cluster_ext_loc].id" -o tsv); do
+              az resource delete -v --id $asset --verbose
+            done
+      - name: Delete asset endpoint profiles
+        uses: azure/cli@v2
+        with:
+          azcliversion: 2.58.0
+          inlineScript: |
+            endpoint_type="Microsoft.DeviceRegistry/assetEndpointProfiles"
+            in_cluster_ext_loc="contains(to_string(extendedLocation.name), '${{ env.CLUSTER_PREFIX }}')"
+            for endpoint in $(az resource list -g ${{ env.RESOURCE_GROUP }} --resource-type $endpoint_type --query "[?$in_cluster_ext_loc].id" -o tsv); do
+              az resource delete -v --id $endpoint --verbose
```

Comment:
you can add a no wait here

(maybe even assets unless you want to make sure we don't get any service requirement surprises)